irohad/consensus/yac/impl/peer_orderer_impl.cpp

```diff
@@ -21,12 +21,18 @@ namespace iroha {
 
       boost::optional<ClusterOrdering> PeerOrdererImpl::getOrdering(
           const YacHash &hash,
-          std::vector<std::shared_ptr<shared_model::interface::Peer>> peers) {
+          std::vector<std::shared_ptr<shared_model::interface::Peer>> const
+              &peers) {
         std::seed_seq seed(hash.vote_hashes.block_hash.begin(),
                            hash.vote_hashes.block_hash.end());
+
+        peer_positions_.resize(peers.size());
+        std::iota(std::begin(peer_positions_), std::end(peer_positions_), 0);
+
         std::default_random_engine gen(seed);
-        std::shuffle(peers.begin(), peers.end(), gen);
-        return ClusterOrdering::create(peers);
+        std::shuffle(peer_positions_.begin(), peer_positions_.end(), gen);
```

Comment:
can you switch this to the platform-independent shuffler please? https://github.com/hyperledger/iroha/pull/336